interactive_templates/templates/v2/analysis/event_variables.py

```diff
@@ -2,82 +2,174 @@
 from utilities import generate_expectations_codes
 
 
-def clinical_event(codelist, date_range, event_name):
-    events = {
-        f"{event_name}": (
-            patients.with_these_clinical_events(
-                codelist=codelist,
-                between=date_range,
-                returning="binary_flag",
-                return_expectations={"incidence": 0.5},
-            )
-        ),
-        f"{event_name}_code": (
-            patients.with_these_clinical_events(
-                codelist=codelist,
-                between=date_range,
-                returning="code",
-                return_expectations={
-                    "rate": "universal",
-                    "category": {"ratios": generate_expectations_codes(codelist)},
-                },
-            )
-        ),
-        f"{event_name}_date": (
-            patients.with_these_clinical_events(
-                codelist=codelist,
-                between=date_range,
-                returning="date",
-                date_format="YYYY-MM-DD",
-                return_expectations={
-                    "date": {
-                        "earliest": "index_date",
-                        "latest": "last_day_of_month(index_date)",
-                    }
-                },
-            )
-        ),
-    }
+def clinical_event(codelist, date_range, event_name, ever=False):
+    """
+    Returns a dictionary of event variables using `with_these_clinical_events` for a given codelist and date range.
+
+    Args:
+        codelist (Codelist): A codelist object.
+        date_range (tuple): A list of two dates in the format YYYY-MM-DD.
+        event_name (str): The name of the event.
+        ever (bool): Whether to overwrite the date range to be on_or_before the end date.
+    """
+    if ever:
+        events = {
+            f"{event_name}": (
+                patients.with_these_clinical_events(
+                    codelist=codelist,
+                    on_or_before=date_range[1],
```

Comment:
There's a lot of duplication just to pick `on_or_before` or `between`.  To avoid that you could use dictionary blatting, where you build up something like this first:

```
if ever:
    date_kwargs = {"on_or_before": date_range[1]}
else:
    date_kwargs = {"between": date_range}
```

then your definitions can look something like this:

```
f"{event_name}": (
    patients.with_these_clinical_events(
        codelist=codelist,
        returning="binary_flag",
        return_expectations={"incidence": 0.5},
        **date_kwargs,
    )
)
````